src/vtktools.jl

```diff
@@ -187,6 +243,96 @@ function calc_node_coordinates(mesh::P4estMesh, n_visnodes)
 end
 
 
+# Calculation of the node coordinates for `TreeMesh` in 2D
+function calc_node_coordinates!(node_coordinates::AbstractArray{<:Any, 4}, nodes, mesh)
```

Comment:
This function seems to be entirely new code. Didn't we need this before? Or is it now duplicated somewhere?

Reply:
No this was not needed before. These routines are analogous to the constructors in Trixi to create the `node_coordinates` for the solver.

Reply:
Couldn't you reuse the functionality from Trixi then? Or use another approach to deduplicate this piece of code? Not saying it should be done in this PR, but if there's an easy way to not have two implementations of essentially the same thing, I'd be great to avoid it (with an issue).

Reply:
This would require a fair amount of restructuring because the `TreeMesh` does not have an explicit `calc_node_coordinates!` function. Instead it is cooked into the `init_elements!` for that mesh type, e.g., see [here](https://github.com/trixi-framework/Trixi.jl/blob/c8f4cd07d0349d8b40891386e4286165371d226d/src/solvers/dgsem_tree/containers_2d.jl#L94) for the 2D version.